tc_aws/loaders/s3_loader.py

```diff
@@ -23,30 +23,91 @@ def load(context, url, callback):
     :param callable callback: Callback method once done
     """
     if _use_http_loader(context, url):
-        http_loader.load_sync(context, url, callback, normalize_url_func=http_loader._normalize_url)
+        http_loader.load_sync(context, url, callback,
+                              normalize_url_func=http_loader._normalize_url)
         return
 
     bucket, key = _get_bucket_and_key(context, url)
 
-    if _validate_bucket(context, bucket):
-        bucket_loader = Bucket(bucket, context.config.get('TC_AWS_REGION'),
-                               context.config.get('TC_AWS_ENDPOINT'))
+    if not _validate_bucket(context, bucket):
+        result = LoaderResult(successful=False,
+                              error=LoaderResult.ERROR_NOT_FOUND)
+        callback(result)
+        return
+
+    bucket_loader = Bucket(bucket, context.config.get('TC_AWS_REGION'),
+                           context.config.get('TC_AWS_ENDPOINT'))
+
+    handle_data = HandleDataFunc.as_func(key,
+                                         callback=callback,
+                                         bucket_loader=bucket_loader,
+                                         context=context)
+
+    bucket_loader.get(key, callback=handle_data)
+
+
+class HandleDataFunc(object):
```

Comment:
Because we need a function for callback, it's hard to wrapper all complexity inside a function. For me, it's look like not cleary. Then, I use a class to have all logical inside it. This help me to test all features.  This is a pattern used at Django for example (Class Based view).

Pro: 
- Easy to test
- Wrapper all logical inside a class 

Negative
- Not use to understand at first look